library/src/main/java/com/cottacush/android/currencyedittext/CurrencyInputWatcher.kt

```diff
@@ -99,8 +102,20 @@ class CurrencyInputWatcher(
         editText.addTextChangedListener(this)
     }
 
-    companion object {
-        val TAG: String = CurrencyInputWatcher::class.java.simpleName
-        var decimalLength = 0
+    /**
+     * @param number the original number to format
+     * @return the appropriate zero sequence for the input number. e.g 156.1 returns "0",
+     *  14.98 returns "00"
+     */
+    private fun getFormatSequenceAfterDecimalSeparator(number: String): String {
+        val noOfCharactersAfterDecimalPoint =
+            number.length - number.indexOf(decimalFormatSymbols.decimalSeparator) - 1
+        val sequence = StringBuilder()
+        var i = 0
+        while (i < noOfCharactersAfterDecimalPoint && i < MAX_NO_OF_DECIMAL_PLACES) {
+            sequence.append("0")
```

Comment:
Nice @r4sh33d . I think we can also do this in one line  using repeat. So we can  Repeat "0"  ```min (noOfCharactersAfterDecimalPoint , MAX_NO_OF_DECIMAL_PLACES)``` times.

Reply:
This makes sense @JohnItoo, I'll update the PR.